src/PlusDataCollection/USDigitalEncodersTracking/vtkPlusUSDigitalEncodersTracker.cxx

```diff
@@ -364,148 +252,94 @@ PlusStatus vtkPlusUSDigitalEncodersTracker::InternalUpdate()
+      tempTransform->Translate(localmovement.GetData());
+    }
+    else //single encoder
+    {
+      // Update transformation matrix of the connected US digital encoder   
+      if (it->Motion == 0)
+      {
+        vtkMath::MultiplyScalar(localmovement.GetData(), encoderValue * it->PulseSpacing);
+        tempTransform->Translate(localmovement.GetData());
+      }
+      else if (it->Motion == 1)
       {
-        long encoderValue;
-        // Get current encoder values from one connected US digital encoder
-        ::A2GetPosition(encoderinfopos->second->Encoder_Addr, &encoderValue);
-
-        // Update transformation matrix of the connected US digital encoder
-        vtkVector3d localmovement = encoderinfopos->second->Encoder_LocalAxis;
-        vtkSmartPointer<vtkTransform> tempTransform = vtkSmartPointer<vtkTransform>::New();
-
-        if (encoderinfopos->second->Encoder_Motion == 0)
-        {
-          vtkMath::MultiplyScalar(localmovement.GetData(),
-                                  encoderValue * encoderinfopos->second->Encoder_PulseSpacing);
-
-          tempTransform->Translate(localmovement.GetData());
-        }
-        else if (encoderinfopos->second->Encoder_Motion == 1)
-        {
-          // Check the unit of rotation angle .... (degree or radian)
-          tempTransform->RotateWXYZ(encoderValue * encoderinfopos->second->Encoder_PulseSpacing,
-                                    localmovement.GetData());
-        }
-        else
-        {
-          LOG_ERROR("Un-supported motion");
-        }
-
-        vtkMatrix4x4::Multiply4x4(encoderinfopos->second->Encoder_TrackingInfo.Encoder_PreTMatrix,
-                                  tempTransform->GetMatrix(),
-                                  encoderinfopos->second->Encoder_TrackingInfo.Encoder_TransformationMatrix);
-
-        this->USDigitalEncoderTransformRepository->SetTransform(encoderinfopos->second->Encoder_TrackingInfo.Encoder_TransformName,
-            encoderinfopos->second->Encoder_TrackingInfo.Encoder_TransformationMatrix);
-        if (ToolTimeStampedUpdateWithvtkPlusEncoderTrackingInfo(encoderinfopos->second->Encoder_TrackingInfo) == PLUS_FAIL)
-        {
-          LOG_ERROR("Unable to find tool on port: " << encoderinfopos->second->Encoder_TrackingInfo.Encoder_PortName);
-          continue;
-        }
+        // Check the unit of rotation angle .... (degree or radian)
+        tempTransform->RotateWXYZ(encoderValue * it->PulseSpacing, localmovement.GetData());
+      }
+      else
+      {
+        LOG_ERROR("Un-supported motion type");
       }
     }
-  }
 
-  for (unsigned int i = 0; i < USDigitalEncoderTrackingInfoList.size(); ++i)
-  {
-    if (!USDigitalEncoderTrackingInfoList[i].Encoder_Persistent)
+    it->TransformationMatrix->DeepCopy(tempTransform->GetMatrix());
+    this->TransformRepository->SetTransform(it->TransformName, it->TransformationMatrix);
+    if (MyToolTimeStampedUpdate(*it) == PLUS_FAIL)

```

Comment:
I admit I don't love the "My" nomenclature as it is not descriptive. I am open to a discussion though as I see it is cleaner. Something in between?